src/fixhorizon.jl

```diff
@@ -25,43 +30,120 @@ const FHWrapper = Union{FixedHorizonMDPWrapper,FixedHorizonPOMDPWrapper}
 end
 
-POMDPs.isterminal(w::FHWrapper, ss::Tuple{<:Any,Int}) = last(ss) > w.horizon || isterminal(w.m, first(ss))
+"""
+    POMDPs.isterminal(w::FHWrapper, ss::Tuple{<:Any,Int})
+
+Mark the state as terminal if its stage number if greater than horizon, else let the Infinite Horizon MDP's `isterminal` method decide.
+"""
+POMDPs.isterminal(w::FHWrapper, ss::Tuple{<:Any,Int}) = stage(ss) > w.horizon || isterminal(w.m, first(ss))
 
 function POMDPs.gen(w::FHWrapper, ss::Tuple{<:Any,Int}, a, rng::AbstractRNG)
     out = gen(w.m, first(ss), a, rng)
     if haskey(out, :sp)
-        return merge(out, (sp=(out.sp, last(ss)+1),))
+        return merge(out, (sp=(out.sp, stage(ss)+1),))
     else
         return out
     end
 end
 
-POMDPs.transition(w::FHWrapper, ss::Tuple{<:Any,Int}, a) = InStageDistribution(transition(w.m, first(ss), a), last(ss)+1)
+"""
+    POMDPs.transition(w::FHWrapper, ss::Tuple{<:Any,Int}, a)
+
+Wrap the transition result of Infinite Horizon MDP with stage number.
+"""
+POMDPs.transition(w::FHWrapper, ss::Tuple{<:Any,Int}, a) = InStageDistribution(transition(w.m, first(ss), a), stage(ss)+1)
 # TODO: convert_s
 
 POMDPs.actions(w::FHWrapper, ss::Tuple{<:Any,Int}) = actions(w.m, first(ss))
 
+"""
+    POMDPs.initialstate(w::FHWrapper)
+
+Wrap Infinite Horizon MDP's `initialstate(mdp)` with stage 1.
+"""
 POMDPs.initialstate(w::FHWrapper) = InStageDistribution(initialstate(w.m), 1)
 
+"""
+    POMDPs.observations(w::FixedHorizonPOMDPWrapper)
+
+Create a product of Infinite Horizon MDP's observations with all not-terminal stages (`1:w.horizon`).
+"""
 POMDPs.observations(w::FixedHorizonPOMDPWrapper) = Iterators.product(observations(w.m), 1:w.horizon)
+
+# TODO: Write Docstring
 POMDPs.obsindex
-POMDPs.observation(w::FixedHorizonPOMDPWrapper, ss::Tuple{<:Any,Int}, a, ssp::Tuple{<:Any, Int}) = InStageDistribution(observation(w.m, first(ss), a, first(ssp)), last(ss))
+
+"""
+    POMDPs.observation(w::FixedHorizonPOMDPWrapper[, ss::Tuple{<:Any,Int}], a, ssp::Tuple{<:Any, Int})
+
+Create a product of Infinite Horizon MDP's observations given destination state and action (and original state) with original state's stage.
+"""
+POMDPs.observation(w::FixedHorizonPOMDPWrapper, ss::Tuple{<:Any,Int}, a, ssp::Tuple{<:Any, Int}) = InStageDistribution(observation(w.m, first(ss), a, first(ssp)), stage(ss))
 POMDPs.observation(w::FixedHorizonPOMDPWrapper, a, ssp::Tuple{<:Any, Int}) = InStageDistribution(observation(w.m, a, first(ssp)), last(ssp)-1)
+
+"""
+    POMDPs.initialstate(w::FHWrapper)
+
+Return Infinite Horizon MDP's initial observations.
+"""
 POMDPs.initialobs(w::FixedHorizonPOMDPWrapper, ss::Tuple{<:Any,Int}) = initialobs(w.m, first(ss))
 # TODO: convert_o
 
 ###############################
 # FiniteHorizonPOMDPs interface
 ###############################
-stage_states(w::FHWrapper, stage::Int) = states(w.m)
+"""
+    stage(ss::Tuple{<:Any,Int})
+
+Return number of state's stage
+"""
+stage(ss::Tuple{<:Any,Int}) = last(ss)
```

Comment:
This should be documented in interface.jl - this docstring seems redundant to me. Same for the functions below.

Reply:
I was not sure how to write docstrings in cases of declaration in the interface and with the definition in `fixhorizon`.
The usage with input parameters in the docstring in the interface is bad, so I wanted to make explicit docstring in `fixhorizon`.
Anyway, I fixed this.